File: C#Part2 - Homework/03. Methods - Homework/04.AppearanceCount/AppearanceCount.cs
namespace AppearanceCount
{
    using System;

    class AppearanceCount
    {
        /* Write a method that counts how many times given number appears in given array.
           Write a test program to check if the method is workings correctly.
         */

        static void Main()
        {
            Console.WriteLine("Enter size of array");
            int size = int.Parse(Console.ReadLine());
            int[] numbers = new int[size];

            for (int i = 0; i < size; i++)
            {
                Console.Write("Index[{0}] = ", i);
                numbers[i] = int.Parse(Console.ReadLine());
            }

            Console.Write("Enter search number = ");
            int searchNum = int.Parse(Console.ReadLine());

            Console.WriteLine("The number {0} is present {1}",searchNum, CountGivenNumber(numbers, searchNum));
        }
        static int CountGivenNumber(int[] num, int search)
        {
            int countNum = 0;
            for (int i = 0; i < num.Length; i++)
            {
                if (search == num[i])
                {
                    countNum++;
                }
            }
            return countNum;
        }
    }
}

File: JavaScript/Homework/Functions - Homework/Problem 5 Appearance count/AppearanceCount.js
// Write a function that counts how many times given number appears in given array.
// Write a test function to check if the function is working correctly.

var btn = document.getElementById('print-nums-button');

function checkHowManyTimes(arr, digit) {
    var count = 0;
    for (var i = 0; i < arr.length; i+=1) {
        if (arr[i] === digit) {
            count+=1;
        }
    };
    return count;
}
btn.addEventListener('click', function () {
    var inputValue = document.getElementById('value').value;
    var givenDigit = document.getElementById('number').value;
    var resultArea = document.getElementById('result');
    var arrayOfDigits = inputValue.split(' ');
    var repeatedTimes = checkHowManyTimes(arrayOfDigits, givenDigit);
    result.innerHTML = "Digit " + givenDigit + " is repeated " + repeatedTimes + " times in the array " + arrayOfDigits;
})